Src/Support/Google.Apis.Auth/OAuth2/DefaultCredentialProvider.cs

```diff
@@ -339,6 +341,28 @@ private static IGoogleCredential CreateExternalCredentialFromParameters(JsonCred
             }
         }
 
+        private ImpersonatedCredential CreateImpersonatedServiceAccountCredentialFromParameters(JsonCredentialParameters parameters)
+        {
+            if (parameters.Type != JsonCredentialParameters.ImpersonatedServiceAccountCredentialType
+                || parameters.SourceCredential is null
+                || string.IsNullOrEmpty(parameters.ServiceAccountImpersonationUrl))
+            {
+                throw new InvalidOperationException("JSON data does not represent a valid impersonated service account credential.");
+            }
+
+            // If source credential is of a credential type that does not support impersonation, attemting to create the
+            // impersonated credential will faill a few lines after.
```

Comment:
faill => fail (and I'd personally use "later" instead of "after" but the meaning is clear)